ctru-rs/examples/ir-user-circle-pad-pro.rs

```diff
@@ -0,0 +1,259 @@
+//! A demo of using the ir:USER service to connect to the Circle Pad Pro.
+
+use ctru::prelude::*;
+use ctru::services::gfx::{Flush, Swap};
+use ctru::services::ir_user::{CirclePadProInputResponse, ConnectionStatus, IrDeviceId, IrUser};
+use ctru::services::svc::HandleExt;
+use ctru_sys::Handle;
+use std::time::Duration;
+
+const PACKET_INFO_SIZE: usize = 8;
+const MAX_PACKET_SIZE: usize = 32;
+const PACKET_COUNT: usize = 1;
+const PACKET_BUFFER_SIZE: usize = PACKET_COUNT * (PACKET_INFO_SIZE + MAX_PACKET_SIZE);
+const CPP_CONNECTION_POLLING_PERIOD_MS: u8 = 0x08;
+const CPP_POLLING_PERIOD_MS: u8 = 0x32;
+
+fn main() {
+    let apt = Apt::new().unwrap();
+    let gfx = Gfx::new().unwrap();
+    let top_console = Console::new(gfx.top_screen.borrow_mut());
+    let bottom_console = Console::new(gfx.bottom_screen.borrow_mut());
+    let mut demo = CirclePadProDemo::new(top_console, bottom_console);
+    demo.print_status_info();
+
+    // Initialize HID after ir:USER because libctru also initializes ir:rst,
+    // which is mutually exclusive with ir:USER. Initializing HID before ir:USER
+    // on New 3DS causes ir:USER to not work.
```

Comment:
Either you add the definition for the weak function, or specify that this example (currently) works only on Old3DS. Otherwise, this comment becomes false and misleading.

Reply:
Added the function export